src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -53,31 +54,35 @@ static zns_sm_metadata_t make_metadata(
     "seastore_device_size");
   
   size_t size = (data.size == 0) ? config_size : data.size;
+  size_t segment_size = zone_size << SECT_SHIFT;
+  size_t zones_per_segment = segment_size / (zone_size << SECT_SHIFT);
+  size_t segments = (num_zones - 1) / zones_per_segment;
```

Comment:
The initial implementation had done it this way. And I think for now, it is better if it is equal to zone size, as zone size is directly mapped to segment size so it makes sense to keep the `first_segment_offset` same as zone size, to keep the offset calculation simple. 
Else, we would need to do special handling for segment/zone 0, so that the write pointer is always incremented by size of the superblock, to get the correct physical WP.